### PR TITLE
bugfix: containerized database url accomodation

### PR DESCRIPTION
**Pull request by nugentmichael1**
- Added another environment variable to the docker-compose.yml file because a containerized spring boot application and containerized Postgres database communicate differently than in non docker/containerized scenarios.
- Changed containerized Postgres instance's port so as not to conflict with any host Postgres installations.
- Requires additional line to .env file: ```DOCKER_DATASOURCE_URL=jdbc:postgresql://db:5432/${POSTGRES_DB}```
- My .env file looks like this: 
```
POSTGRES_DB=p2team1
POSTGRES_USER=p2team1_user
POSTGRES_PASSWORD=password

SPRING_DATASOURCE_USERNAME=${POSTGRES_USER}
SPRING_DATASOURCE_PASSWORD=${POSTGRES_PASSWORD}
SPRING_DATASOURCE_URL=jdbc:postgresql://localhost:5432/${POSTGRES_DB}
SPRING_API_URL=http://localhost:8080

DOCKER_DATASOURCE_URL=jdbc:postgresql://db:5433/${POSTGRES_DB}
```
- In summary, this enables someone to run the backend application via the IntelliJ IDE (requires local Postgres DB) or the terminal's ```docker compose up --build``` command.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,20 +15,27 @@ See [Spring Boot Database Initialization Documentation](https://docs.spring.io/s
...
 ```
+# Assign values to these three variables with information from your local Postgres DB installation:
 POSTGRES_DB=p2team1
 POSTGRES_USER=p2team1_user
 POSTGRES_PASSWORD=password
 
 SPRING_DATASOURCE_USERNAME=${POSTGRES_USER}
 SPRING_DATASOURCE_PASSWORD=${POSTGRES_PASSWORD}
 SPRING_DATASOURCE_URL=jdbc:postgresql://localhost:5432/${POSTGRES_DB}
+# Not entirely sure what this does, yet.  Could change.
```
Comment (treycrossley) [inline README.md:29]: its literally just shorthand for the API URL in the app, you could hardcode this if you want. Its only used in the DataInitializer rn

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-spring.datasource.url=${SPRING_DATASOURCE_URL}
-spring.datasource.username=${SPRING_DATASOURCE_USERNAME}
-spring.datasource.password=${SPRING_DATASOURCE_PASSWORD}
-#####
-# Hibernate configurations
+
+##### Datasource configurations
+##### These should be auto-detected through the environment variables' names, but uncomment in case they are not (like maybe in production)
+#spring.datasource.url=${SPRING_DATASOURCE_URL}
+#spring.datasource.username=${SPRING_DATASOURCE_USERNAME}
```
Comment (treycrossley) [inline src/main/resources/application.properties:6]: OH theyre autoconfigured?! thats dope!

**Review by treycrossley — APPROVED** [review]
